microcosm_daemon/runner.py

```diff
@@ -35,14 +36,35 @@ def __init__(self, processes, target, *args, **kwargs):
         self.target = target
         self.args = args
         self.kwargs = kwargs
+        self.pool = None
+
+        self.init_signal_handlers()
 
     def run(self):
-        pool = Pool(processes=self.processes)
+        self.pool = Pool(processes=self.processes)
+
         for _ in range(self.processes):
-            pool.apply_async(_start, (self.target,) + self.args, self.kwargs)
+            self.pool.apply_async(_start, (self.target,) + self.args, self.kwargs)
+
+        self.close()
+
+    def init_signal_handlers(self):
+        for signum in (SIGINT, SIGTERM):
+            signal(signum, self.on_terminate)
+
+    def close(self, terminate=False):
+        if self.pool is not None:
+            if terminate:
+                self.pool.terminate()
+            else:
+                self.pool.close()
+
+            try:
+                self.pool.join()
+            except KeyboardInterrupt:
```

Comment:
is this reachable any more now that you are handing SIGINT explicitly?

Reply:
👍